pkg/ipamd/ipamd.go

```diff
@@ -1353,8 +1355,11 @@ func podENIErrInc(fn string) {
 
 // nodeIPPoolReconcile reconcile ENI and IP info from metadata service and IP addresses in datastore
 func (c *IPAMContext) nodeIPPoolReconcile(ctx context.Context, interval time.Duration) {
+	// To reduce the number of EC2 API calls, skip reconciliation if IPs were recently added to the datastore.
 	timeSinceLast := time.Since(c.lastNodeIPPoolAction)
-	if timeSinceLast <= interval {
+	// Make an exception if node needs a trunk ENI and one is not currently attached.
```

Comment:
One callout for this change is that, if trunk ENI fails to attach for any VPC RC issues then aws-node keeps running the reconciliation..maybe as an enhancement we should make it like a backoff retry in cases of SGPP..